### ISSUE TITLE
RESULT_CODE_KILLED_BAD_MESSAGE 

### ISSUE THREAD

**Issue by OneSeven** [MicrosoftEdge/WebView2Feedback#1968]
**Description**
```html
<video id="videoElement" controls></video>
```
```js
<script src="https://cdn.jsdelivr.net/npm/mpegts.js@1.6.10/dist/mpegts.js"></script>
if (mpegts.getFeatureList().mseLivePlayback) {
        let videoElement = document.getElementById('videoElement');
        let player = mpegts.createPlayer({
            type: 'flv',  // could also be mpegts, m2ts, flv
            isLive: true,
            url: 'http://localhost:7001/live/movie.flv',
            cors: true,
        }, {
            enableWorker: true,
            enableStashBuffer: false,
            stashInitialSize: 128
        });
        player.attachMediaElement(videoElement);
        player.load();
    }
```
Running the above code crashes
Normal in the same version of "edge"
Normal before 95 version

**Version**
SDK: 
Runtime: 96.0.1054.34 
Framework: 
OS: windows 10

**Repro Steps**
Running the above code crashes

**Screenshots**
![zzzzzzzzzzzzzzz](https://user-images.githubusercontent.com/17974392/143675562-ee560703-cbb3-48d1-beda-c4e75ef0806a.png)
**Additional context**

**Comment by champnic** [comment]
Thanks for the bug report @OneSeven - I've added it to our backlog and we're taking a look.

**Comment by champnic** [comment]
@OneSeven We've seen similar issues to this in the past that were caused by antivirus software. Can you try your scenario again after disabling any antivirus software on the machine to see if that might be the cause here as well?

**Comment by OneSeven** [comment]
@champnic There is only windows defender on my computer. I turned off windows defender and tested it. The result is still the crash. It is normal in windows edge, and it is normal in webview2 runtime 95 and earlier versions. I can’t understand this. Please troubleshoot the discrepancies

**Comment by champnic** [comment]
Thanks for verifying! Would you be able to grab a netlog of this scenario and share it? You can email the log to me at `champnic@microsoft.com` if you'd like.
https://textslashplain.com/2020/01/17/capture-network-logs-from-edge-and-chrome/

**Comment by champnic** [comment]
Hey @OneSeven - I just gave this a try and didn't experience a crash. What runtime version are you seeing this on?

**Comment by champnic** [comment]
Are you running in a Service Worker?
<img width="552" alt="image" src="https://user-images.githubusercontent.com/45700886/144520766-77d01ce6-4c3a-4e1f-8bbf-13eadbd36d66.png">

**Comment by OneSeven** [comment]
> Hey @OneSeven - I just gave this a try and didn't experience a crash. What runtime version are you seeing this on?

96.0.1054.34 and 96.0.1054.41

**Comment by OneSeven** [comment]
> Are you running in a Service Worker?
> <img width="552" alt="image" src="https://user-images.githubusercontent.com/45700886/144520766-77d01ce6-4c3a-4e1f-8bbf-13eadbd36d66.png">
> 

No, it will crash if two lines of code are executed directly in the script

**Comment by champnic** [comment]
Hm, I tried in the debugger and it worked ok on 96.0.1054.34:
<img width="379" alt="image" src="https://user-images.githubusercontent.com/45700886/144527977-91b31bb9-93fe-4aef-9ae9-38095dfabe24.png">

If you try that in the debugger in either a WebView2 or the Edge browser does it still crash? And when you say crash, do you mean "have an error and display the RESULT... error page"? or the WebView2 process is killed?

**Comment by OneSeven** [comment]
> Hm, I tried in the debugger and it worked ok on 96.0.1054.34: <img alt="image" width="379" src="https://user-images.githubusercontent.com/45700886/144527977-91b31bb9-93fe-4aef-9ae9-38095dfabe24.png">
> 
> If you try that in the debugger in either a WebView2 or the Edge browser does it still crash? And when you say crash, do you mean "have an error and display the RESULT... error page"? or the WebView2 process is killed?

Execution in the console will also crash,
I am building in this project https://github.com/wailsapp/wails
https://github.com/wailsapp/wails/discussions/992

![mmmmm](https://user-images.githubusercontent.com/17974392/144528446-f2d24ea0-6fc6-457b-bd46-f36272e0661a.png)

**Comment by OneSeven** [comment]
> Hm, I tried in the debugger and it worked ok on 96.0.1054.34: <img alt="image" width="379" src="https://user-images.githubusercontent.com/45700886/144527977-91b31bb9-93fe-4aef-9ae9-38095dfabe24.png">
> 
> If you try that in the debugger in either a WebView2 or the Edge browser does it still crash? And when you say crash, do you mean "have an error and display the RESULT... error page"? or the WebView2 process is killed?

The edge is normal, but the webview crashes
![mmm](https://user-images.githubusercontent.com/17974392/144528738-1f248919-edd5-4c59-b401-d71f9f047b57.png)

**Comment by OneSeven** [comment]
> 

Maybe we can perform remote debugging through Anydesk

**Comment by leaanthony** [comment]
@champnic - Hey Nic 😄. Let me know what I can do on my side (Wails uses the win32 webview2 component) to help debug this.  Many thanks.

**Comment by champnic** [comment]
@leaanthony Are you able to reproduce the issue @OneSeven is mentioning above? Is Wails doing anything with permissions, such as denying any of them?

**Comment by leaanthony** [comment]
I will try and reproduce today, but I believe that by default all permissions are granted and no cli flags are set. If there are specific permissions to check for, I can do that. I'm just not familiar with this JS API.

**Comment by OneSeven** [comment]
@leaanthony @champnic 

Let me explain the phenomenon. The program built with wails half a month ago is normal. After half a month, the program built at the beginning will also have problems. During this half month, I upgraded the win10 system, .net The framework has edge and webview runtime, I guess it may not have much to do with wails

**Comment by leaanthony** [comment]
Sounds like a runtime regression

**Comment by OneSeven** [comment]
> Sounds like a runtime regression

How can I install the old version of webview runtime? I can't find the installation file for version 95 now

**Comment by champnic** [comment]
Agreed this sounds like a runtime regression. @OneSeven If you look in the `<user data folder>\EBWebView\crashpad\reports` do you see any dumps in there? If so, would you be able to share with me at `champnic@microsoft.com` (may need to upload to a cloud sharing service as they can be large)?

**Comment by OneSeven** [comment]
> Agreed this sounds like a runtime regression. @OneSeven If you look in the `<user data folder>\EBWebView\crashpad\reports` do you see any dumps in there? If so, would you be able to share with me at `champnic@microsoft.com` (may need to upload to a cloud sharing service as they can be large)?

I have sent two sets of program data to you, please check

**Comment by champnic** [comment]
Thanks! I took a look and it seems to be failing here:
```cpp
void BlobRegistryImpl::URLStoreForOrigin(
    const url::Origin& origin,
    mojo::PendingAssociatedReceiver<blink::mojom::BlobURLStore> receiver) {
  Delegate* delegate = receivers_.current_context().get();
  DCHECK(delegate);
  if (!origin.opaque() && !delegate->CanCommitURL(origin.GetURL())) {
    mojo::ReportBadMessage(
        "Non committable origin passed to BlobRegistryImpl::URLStoreForOrigin");
```

Can you share what the value of `mediaSourc` is in your example that's causing the failure?
```js
let mediaSourc = new window.MediaSource();
let mediaSourceObjectURL = window.URL.createObjectURL(mediaSourc);
```

**Comment by OneSeven** [comment]
> Thanks! I took a look and it seems to be failing here:
> 
> ```c++
> void BlobRegistryImpl::URLStoreForOrigin(
>     const url::Origin& origin,
>     mojo::PendingAssociatedReceiver<blink::mojom::BlobURLStore> receiver) {
>   Delegate* delegate = receivers_.current_context().get();
>   DCHECK(delegate);
>   if (!origin.opaque() && !delegate->CanCommitURL(origin.GetURL())) {
>     mojo::ReportBadMessage(
>         "Non committable origin passed to BlobRegistryImpl::URLStoreForOrigin");
> ```
> 
> Can you share what the value of `mediaSourc` is in your example that's causing the failure?
> 
> ```js
> let mediaSourc = new window.MediaSource();
> let mediaSourceObjectURL = window.URL.createObjectURL(mediaSourc);
> ```

Just create an empty MediaSource and createObjectURL on it, it will crash 
Just the two lines of code above
You are in an empty project, no need to write anything, just add two lines of code, it will crash

**Comment by champnic** [comment]
Per my comment above, I've already tried those lines and they don't crash for me:
https://github.com/MicrosoftEdge/WebView2Feedback/issues/1968#issuecomment-985132395

As I'm unable to repro, I can't see what `mediaSourc` value might be happening on your machine that might be causing the failure in `createObjectURL`.

**Comment by OneSeven** [comment]
I build by wails
@leaanthony Please help with joint debugging 
@champnic Please communicate with leaanthony

**Comment by OneSeven** [comment]
> Per my comment above, I've already tried those lines and they don't crash for me: [#1968 (comment)](https://github.com/MicrosoftEdge/WebView2Feedback/issues/1968#issuecomment-985132395)
> 
> As I'm unable to repro, I can't see what `mediaSourc` value might be happening on your machine that might be causing the failure in `createObjectURL`.

I still want to remind that he works normally in the webview rumtime 95 version, do I need to rule out the direct difference between 96 and 95?

**Comment by champnic** [comment]
@OneSeven Are you able to get a WebView2 launched in your scenario (maybe to a separate page) and then try running this line manually to see what value it's giving you?
`let mediaSourc = new window.MediaSource();`

**Comment by OneSeven** [comment]
> @OneSeven Are you able to get a WebView2 launched in your scenario (maybe to a separate page) and then try running this line manually to see what value it's giving you? `let mediaSourc = new window.MediaSource();`

There is no error in performing this step,
Error when running window.URL.createObjectURL(mediaSourc)
![屏幕截图 2021-12-08 095622](https://user-images.githubusercontent.com/17974392/145134719-558edc5d-e66c-48a8-84ee-eafe9aaa40ce.png)

**Comment by champnic** [comment]
@OneSeven I noticed you are doing this from a local file. Can you try running these steps from a website?

**Comment by champnic** [comment]
I tried with a local file and just using NavigateToString and wasn't able to repro, so I'm not sure if it's related or not.

**Comment by OneSeven** [comment]
@jchv Please pay attention to this question, wails are based on go-webview2

**Comment by OneSeven** [comment]
> I tried with a local file and just using NavigateToString and wasn't able to repro, so I'm not sure if it's related or not.

I may not be able to provide other information. After I build and suspend the webview through wails, I will report errors when running those two lines of code in the webview's console.
